ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/BlockFetch/Server.hs

```diff
@@ -107,17 +107,18 @@ blockFetchServer _tracer chainDB _version registry = senderSide
         -- When we got an iterator, it will stream at least one block since
         -- its bounds are inclusive, so we don't have to check whether the
         -- iterator is empty.
-        Right it -> SendMsgStartBatch $ sendBlocks it
+        Right it -> SendMsgStartBatch $ sendBlocks 0 it
 
-    sendBlocks :: ChainDB.Iterator m blk (WithPoint blk (Serialised blk))
+    sendBlocks :: Int -> ChainDB.Iterator m blk (WithPoint blk (Serialised blk))
                -> m (BlockFetchSendBlocks (Serialised blk) (Point blk) m ())
-    sendBlocks it = do
+    sendBlocks blockCount it = do
```

Comment:
Shouldn't this be `!blockCount`?

Reply:
Fixed.  Thanks!

Reply:
I think we shouldn't do the counting in ChainSync server. That's an example of something that should happen downstream, in the metric logic that processes the "I served a block" messages. (See my similar comment here https://github.com/input-output-hk/ouroboros-network/pull/2874#issuecomment-762549330 )